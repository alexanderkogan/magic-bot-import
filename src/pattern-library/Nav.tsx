import * as React from 'react'
import { Props } from '../react-typings'

export default (props: Props) => (
    <nav>
        <ul className="obc_tabs">{props.children}</ul>
    </nav>
)
