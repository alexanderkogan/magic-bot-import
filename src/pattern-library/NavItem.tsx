import * as React from 'react'
import { Props } from '../react-typings'
import { NavLink } from 'react-router-dom'

export interface TabItemProps {
    to: string
}
export default (props: Props<TabItemProps>) => (
    <li className="obc_tabs__item">
        <NavLink to={props.to} className="obc_tabs__link" activeClassName="obc_tabs__link--selected" exact>
            {props.children}
        </NavLink>
    </li>
)
