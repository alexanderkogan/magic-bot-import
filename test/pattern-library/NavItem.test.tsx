import * as React from 'react'

import { BrowserRouter as Router } from 'react-router-dom'
import Nav from '../../src/pattern-library/Nav'
import NavItem from '../../src/pattern-library/NavItem'
import { mount } from '../test-utils'

describe('NavItem', () => {
    it('renders shallow', () => {
        const element = mount(
            <Router>
                <Nav>
                    <NavItem to="some/location">some location</NavItem>
                    <NavItem to="">currentPage</NavItem>
                </Nav>
            </Router>
        )
        expect(element).toMatchSnapshot()
    })
})
