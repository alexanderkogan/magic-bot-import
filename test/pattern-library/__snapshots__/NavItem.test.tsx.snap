// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem renders shallow 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Component>
      <nav>
        <ul
          className="obc_tabs"
        >
          <Component
            to="some/location"
          >
            <li
              className="obc_tabs__item"
            >
              <NavLink
                activeClassName="obc_tabs__link--selected"
                className="obc_tabs__link"
                exact={true}
                to="some/location"
              >
                <Link
                  aria-current={null}
                  className="obc_tabs__link"
                  to={
                    Object {
                      "hash": "",
                      "pathname": "/some/location",
                      "search": "",
                      "state": null,
                    }
                  }
                >
                  <LinkAnchor
                    aria-current={null}
                    className="obc_tabs__link"
                    href="/some/location"
                    navigate={[Function]}
                  >
                    <a
                      aria-current={null}
                      className="obc_tabs__link"
                      href="/some/location"
                      onClick={[Function]}
                    >
                      some location
                    </a>
                  </LinkAnchor>
                </Link>
              </NavLink>
            </li>
          </Component>
          <Component
            to=""
          >
            <li
              className="obc_tabs__item"
            >
              <NavLink
                activeClassName="obc_tabs__link--selected"
                className="obc_tabs__link"
                exact={true}
                to=""
              >
                <Link
                  aria-current="page"
                  className="obc_tabs__link obc_tabs__link--selected"
                  style={Object {}}
                  to={
                    Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": null,
                    }
                  }
                >
                  <LinkAnchor
                    aria-current="page"
                    className="obc_tabs__link obc_tabs__link--selected"
                    href="/"
                    navigate={[Function]}
                    style={Object {}}
                  >
                    <a
                      aria-current="page"
                      className="obc_tabs__link obc_tabs__link--selected"
                      href="/"
                      onClick={[Function]}
                      style={Object {}}
                    >
                      currentPage
                    </a>
                  </LinkAnchor>
                </Link>
              </NavLink>
            </li>
          </Component>
        </ul>
      </nav>
    </Component>
  </Router>
</BrowserRouter>
`;
